docs/stable/core_extensions/iceberg/iceberg_rest_catalogs.md

```diff
@@ -0,0 +1,144 @@
+| `SECRET`                     | `VARCHAR`  | `NULL`   | Name of secret used to communicate with the REST Catalog |
+| `CLIENT_ID`                  | `VARCHAR`  | `NULL`   | CLIENT_ID used for Secret                                     |
+| `CLIENT_SECRET`              | `VARCHAR`  | `NULL`   | CLIENT_SECRET needed for Secret                               |
+| `DEFAULT_REGION`             | `VARCHAR`  | `NULL`   | A Default region to use when communicating with the storage layer |
+| `OAUTH2_SERVER_URI`          | `VARCHAR`  | `NULL`   | OAuth2 server url for getting a Bearer Token                  |
+| `AUTHORIZATION_TYPE`         | `VARCHAR`  | `OAUTH2` | Pass `SigV4` for Catalogs the require SigV4 authorization |
+
+
+The following options can only be passed to a `CREATE SECRET` statement, and they require `AUTHORIZATION_TYPE` to be `OAUTH2`
+
+| Parameter                    | Type       | Default  | Description                                                |
+| ---------------------------- | ---------- | -------- | ---------------------------------------------------------- |
+| `OAUTH2_GRANT_TYPE`          | `VARCHAR`  | `NULL` | Grant Type when requesting an OAuth Token |
+| `OAUTH2_SCOPE`               | `VARCHAR`  | `NULL` | Requested scope for the returned OAuth Access Token |
+
+## Specific Catalog Examples 
+
+### R2 Catalog
+
+To attach to an [R2 cloudflare](https://developers.cloudflare.com/r2/data-catalog/) managed catalog follow the attach steps below. 
+
+
+```sql
+CREATE SECRET r2_secret (
+    TYPE ICEBERG,
+    TOKEN '⟨r2_token⟩'
+);
+
+```
+
+You can create a token by following the [create an API token](https://developers.cloudflare.com/r2/data-catalog/get-started/#3-create-an-api-token) steps in getting started.
+
+Then, attach the catalog with the following commands.
+
+```sql
+ATTACH '⟨warehouse⟩' AS my_r2_catalog (
+    TYPE ICEBERG,
+    ENDPOINT '⟨catalog-uri⟩'
+);
+```
+
+The variables for `warehouse` and `catalog-uri` will be available under the settings of the desired R2 Object Storage Catalog (R2 Object Store > Catalog name > Settings).
+
+### Polaris
+
+To attach to a [Polaris](https://polaris.apache.org) catalog the following commands will work.
+
+```sql
+CREATE SECRET polaris_secret (
+    TYPE ICEBERG,
+    CLIENT_ID '⟨admin⟩',
+    CLIENT_SECRET '⟨password⟩',
+);
+```
+
+```sql
+ATTACH 'quickstart_catalog' as polaris_catalog (
+    TYPE ICEBERG,
+    ENDPOINT '⟨polaris_rest_catalog_endpoint⟩'
+);
+```
+
+
+### Lakekeeper
+
+To attach to a [Lakekeeper](https://docs.lakekeeper.io) catalog the following commands will work.
+
+```sql
+CREATE SECRET lakekeeper_secret (
+    TYPE ICEBERG,
+    CLIENT_ID '⟨admin⟩',
+    CLIENT_SECRET '⟨password⟩',
+    OAUTH2_SCOPE '⟨scope⟩',
+    OAUTH2_SERVER_URI '⟨lakekeeper_oauth_url⟩'
+);
+```
+
+```sql
+ATTACH '⟨warehouse⟩' as lakekeeper_catalog (
+    TYPE ICEBERG,
+    ENDPOINT '⟨lakekeeper_irc_url⟩',
+    SECRET lakekeeper_secret
+);
+```
+
+## Limitations
```

Comment:
nit: wydt of adding a sentence about catalog refreshing not being supported right now. the work around is reattach

similar to https://github.com/duckdb/duckdb-iceberg/issues/247